packages/web/src/lib/web-mode.ts

```diff
@@ -142,23 +143,44 @@ export class WebMode {
 		}
 
 		txns = algosdk.assignGroupID(txns);
+
 		const binaryTxs = txns.map((txn: Transaction) => {
 			return txn.toByte();
 		});
+
 		const base64Txs = binaryTxs.map((txn: Uint8Array) => {
 			return this.algoSigner.encoding.msgpackToBase64(txn);
 		});
-		const toBeSignedTxns = base64Txs.map((txn: string) => {
-			return { txn: txn };
+
+		const toBeSignedTxns = base64Txs.map((txn: string, txnId: number) => {
+			return execParams[txnId].sign === types.SignType.LogicSignature
+				? { txn: txn, signers: [] }
+				: { txn: txn };
 		});
+
 		const signedTxn = await this.signTransaction(toBeSignedTxns);
+
+		// sign smart signature transaction
+		for (let i = 0; i < txns.length; ++i) {
```

Comment:
Just a note. Try to use below method for iterating to array.

`for (let profile of myArray) {
    console.log(profile.name) // profile is object in myArray
}`

This looks more clean. The advantage of this is that we don't have to calculate the length of array  on every iteration which is done here: `i < txns.length`

Reply:
done